server/golden/validator.go

```diff
@@ -0,0 +1,297 @@
+func (v Validator) NewConnection(c *mysql.Conn) {
+	return
+}
+
+func (v Validator) ComInitDB(c *mysql.Conn, schemaName string) error {
+	if err := v.handler.ComInitDB(c, schemaName); err != nil {
+		return err
+	}
+	return v.golden.ComInitDB(c, schemaName)
+}
+
+// ComPrepare parses, partially analyzes, and caches a prepared statement's plan
+// with the given [c.ConnectionID].
+func (v Validator) ComPrepare(c *mysql.Conn, query string) ([]*query.Field, error) {
+	return nil, fmt.Errorf("ComPrepare unsupported")
+}
+
+func (v Validator) ComStmtExecute(c *mysql.Conn, prepare *mysql.PrepareData, callback func(*sqltypes.Result) error) error {
+	return fmt.Errorf("ComStmtExecute unsupported")
+}
+
+func (v Validator) ComResetConnection(c *mysql.Conn) {
+	return
+}
+
+// ConnectionClosed reports that a connection has been closed.
+func (v Validator) ConnectionClosed(c *mysql.Conn) {
+	v.handler.ConnectionClosed(c)
+	v.golden.ConnectionClosed(c)
+}
+
+func (v Validator) ComMultiQuery(
+	c *mysql.Conn,
+	query string,
+	callback func(*sqltypes.Result, bool) error,
+) (string, error) {
+	ag := newResultAggregator(callback)
+	var remainder string
+	eg, _ := errgroup.WithContext(context.Background())
+	eg.Go(func() (err error) {
+		remainder, err = v.handler.ComMultiQuery(c, query, ag.processResults)
+		return
+	})
+	eg.Go(func() error {
+		// ignore errors from MySQL connection
+		_, _ = v.golden.ComMultiQuery(c, query, ag.processGoldenResults)
+		return nil
+	})
+
+	err := eg.Wait()
+	if err != nil {
+		return "", err
+	}
+	ag.compareResults(v.getLogger(c).WithField("query", query))
+
+	return remainder, nil
+}
+
+// ComQuery executes a SQL query on the SQLe engine.
+func (v Validator) ComQuery(
+	c *mysql.Conn,
+	query string,
+	callback func(*sqltypes.Result, bool) error,
+) error {
+	ag := newResultAggregator(callback)
+	eg, _ := errgroup.WithContext(context.Background())
+	eg.Go(func() error {
+		return v.handler.ComQuery(c, query, ag.processResults)
+	})
+	eg.Go(func() error {
+		// ignore errors from MySQL connection
+		_ = v.golden.ComQuery(c, query, ag.processGoldenResults)
+		return nil
+	})
+
+	err := eg.Wait()
+	if err != nil {
+		return err
+	}
+	ag.compareResults(v.getLogger(c).WithField("query", query))
+	return nil
+}
+
+// WarningCount is called at the end of each query to obtain
+// the value to be returned to the client in the EOF packet.
+// Note that this will be called either in the context of the
+// ComQuery resultsCB if the result does not contain any fields,
+// or after the last ComQuery call completes.
+func (v Validator) WarningCount(c *mysql.Conn) uint16 {
+	return 0
+}
+
+func (v Validator) getLogger(c *mysql.Conn) *logrus.Entry {
+	return logrus.NewEntry(v.logger).WithField(
+		sql.ConnectionIdLogField, c.ConnectionID)
+}
+
+type aggregator struct {
+	results  []*sqltypes.Result
+	golden   []*sqltypes.Result
+	callback func(*sqltypes.Result, bool) error
+}
+
+const maxRows = 1024
```

Comment:
Same as my `count` hidden system variable suggestion, if this were also a hidden sysvar then it could be tweakable during runtime. Especially for validation purposes, I think there could be a lot of value in being able to define custom upper limits (or remove the limits altogether with a negative).